docs/references/nextjs/clerk-middleware.mdx

```diff
@@ -387,5 +403,29 @@ When providing `CLERK_ENCRYPTION_KEY`, it is recommended to use a 32-byte (256-b
 openssl rand --hex 32
 ```
 
+For multi-tenant applications, you can dynamically define Clerk keys depending on the incoming request, as shown in the following example:
+
+```ts {{ filename: 'middleware.ts' }}
+import { clerkMiddleware } from '@clerk/nextjs/server'
+
+const tenantKeys = {
+  tenant1: { publishableKey: 'pk_tenant1...', secretKey: 'sk_tenant1...' },
+  tenant2: { publishableKey: 'pk_tenant2...', secretKey: 'sk_tenant2...' },
+}
+
+export default clerkMiddleware(
+  (auth, req) => {
+    // Add your middleware checks
+  },
+  (req) => {
+    // Resolve tenant based on the request
+    const tenant = getTenant(req)
+
+    // You would typically fetch these keys from a external store or environment variables.
```

Comment:
Should this comment go above the `const tenantKeys = {` line ?

Reply:
Yeah, I think it makes more sense to move it there.